src/js/components/Form/Form.js

```diff
@@ -0,0 +1,129 @@
+import React, { Component } from 'react';
+import { defaultProps } from '../../default-props';
+import { deepMerge } from '../../utils';
+import { FormContext } from './FormContext';
+
+const defaultMessages = {
+  invalid: 'invalid',
+  required: 'required',
+};
+
+class Form extends Component {
+  static defaultProps = {
+    messages: defaultMessages,
+    value: {},
+  };
+
+  static getDerivedStateFromProps(nextProps, prevState) {
+    const { value, errors, messages } = nextProps;
+    const {
+      value: stateValue,
+      errors: stateErrors,
+      priorValue,
+      priorErrors,
+      priorMessages,
+    } = prevState;
+    if (
+      !priorValue ||
+      value !== priorValue ||
+      errors !== priorErrors ||
+      messages !== priorMessages
+    ) {
+      return {
+        value: value !== priorValue ? value : stateValue,
+        priorValue: value,
+        errors: (errors !== priorErrors ? errors : stateErrors) || {},
+        priorErrors: errors,
+        messages: deepMerge(defaultMessages, messages),
+        priorMessages: messages,
+      };
+    }
+    return null;
+  }
+
+  state = { errors: {}, value: {}, touched: {} };
+
+  validations = {};
+
+  onSubmit = event => {
+    const { onSubmit } = this.props;
+    const { errors, value } = this.state;
+    event.preventDefault();
```

Comment:
can we add a comment explaining why event.preventDefault is here?

Reply:
done